### ISSUE TITLE
Type declarations

### ISSUE THREAD

**Issue by rsgilbert** [rsgilbert/navodata#5]
Lack of type support. Need to properly add declarations so they can be used by VSCode intellisense

**Comment by devgar** [comment]
I could work on this,

As I know, to define types can use .d.ts files or directly typescript. I'm with using typescript, although it implies rewriting existing code and setting a build workflow, this project is still small and is better to set up TS now than after being bigger and more complex.

What do you think?

**Comment by rsgilbert** [comment]
Yeah, typescript would be good. The advantage is that it will be easier to use the project in other typescript projects since the types will be well defined and visible. The disadvantage of course is that its more ambitious. 

If you are willing to refactor the app to typescript I'll happily contribute. There are a number of tests already in place which we can use to ensure we dont break functionality.

Are you familiar with odata or NAV/BC ?

**Comment by devgar** [comment]
> If you are willing to refactor the app to typescript I'll happily contribute. 

Ok, I'll start TS refactor.

> Are you familiar with odata or NAV/BC ?

Yes, I started developing BC extensions this year and I found this repo looking for tools to work with BC API REST.

**Comment by rsgilbert** [comment]
Hello Edgar,

Any update on the navodata refactor?

Sincerely,

Gilbert.

**Comment by rsgilbert** [comment]
Hello @devgar , should we close this?
